### ISSUE TITLE
can't find my data written on qsfs dir after destroy qsfs and redeploy 

### ISSUE THREAD

**Issue by mhost39** [threefoldtech/zos#1444]
using devnet and node 12
using terraform provider: i deployed a QSFS with zdb 10G and cache 10G

after deployed i downloaded 354M on /qsfs:
```
wget https://cloud-images.ubuntu.com/bionic/current/bionic-server-cloudimg-amd64.img
```
then killed the deployment (qsfs and the container and keep the zdbs) and redeploy it
i should find my data after deploy again 
i can't find any thing in /qsfs

**Comment by rkhamis** [comment]
@mhost39 can you provide configuration of zstore?

**Comment by OmarElawady** [comment]
I tried the same scenario and the qsfs container is missing the `cp` binary and `/tmp` mount. But I added them manually (and flushed all nft just in case) and zstor failed to push the data.

zstor container log: https://gist.github.com/OmarElawady/2b4168d6cec9de72ca1cf3fb105bc76c
zstor config: https://gist.github.com/OmarElawady/a5f3c18ebd66af2f8df4fca3ea32c1ed

**Comment by mhost39** [comment]
@rkhamis 
https://gist.github.com/mhost39/dd11a5a005220d72daeee35e52c2a5c7

**Comment by LeeSmet** [comment]
0-stor refuses to save the file, because of the setting of "root" vs the file path. if the `root` config option is set, 0-stor only uploads files which are part of this directory tree (and strips the prefix from the actual path when generating the key).If a file is loaded, this path is prefixed to the path used to upload. Example

On machine 1, set `root` to `/a`, and upload file `/a/b/c`. 0-stor will upload the file, strip `/a`, then hash it to generate the key (so the key is `/b/c`). On a second machine, set `root` to `/d`. Now, the file can be recovered as `/d/b/c`. 0-stor will remove the root from the path again before hashing it to generate the key (thus the hash input is again `/b/c`), which will cause it to find the prior uploaded file.

**Comment by xmonader** [comment]
@mhost39 this should be in verification now right?

**Comment by mhost39** [comment]
@xmonader  not fixed yet

**Comment by OmarElawady** [comment]
I tried locally the flow with the qsfs [container](https://hub.grid.tf/omar0.3bot/qsfs.flist) with all latest binaries. I wrote 400MB random data and 4 85M shards were uploaded (I guess the rest were not rotated yet). But after killing the container and resurrecting it, the mount dir was empty and the logs shows no calls with "missing-data".

Restarting the container removes everything, the zdbfs and zdb data (the one with the hook).

The docker init command: `docker run -ti --rm --name qsfs --privileged  --device /dev/fuse  --mount type=bind,source=/qsfs,target=/mnt,bind-propagation=rshared qsfs /sbin/zinit init`
